WIP podresources: expose API to get all resources

ffromani opened the pull request:
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Add a new endpoint to let client enumerate all the devices known
to the kubelet; this allows clients to know about worker node
capacity.

This change enables topology-aware scheduling components to
report to the scheduler the worker node resource capacity and
allocation, using the topology identifiers (currently NUMA node)
to properly partition the resources.

**Special notes for your reviewer**:
Rationale for this PR is in https://github.com/kubernetes/enhancements/pull/1969

**Does this PR introduce a user-facing change?**:
```release-note
Add endpoint to podresources API to let client applications enumerate the devices known by the kubelet
```

ffromani committed:
podresources: add GetAllocatableResources

Add a new endpoint to let client enumerate all the resources -
cpus + devices - known to the kubelet;
this allows clients to know about worker node capacity.

This change enables topology-aware scheduling components to
report to the scheduler the worker node resource capacity and
allocation, using the topology identifiers (currently NUMA node)
to properly partition the resources.

Note about e2e tests:
Any device managed by deviceplugin is OK here. sr-iov devices
were used only because they are easily available.
Nothing prevents us to just use different devices in the future,
there no specific SR-IOV dependency here.

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
/hold
missing KEP reference (coming ASAP)
I want to doublecheck I didn't lost any change introduced in 28768166f5c8ee179d21d9eb904c3eb4502db7fd when moving code around

ffromani committed:
cpumanager: get online cpu from sysfs

The cadvisor don't expose the online cpu set, thus
we need to fetch if from sysfs.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
TMP: podresources proto changes

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
TMP: podresources generated file

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
replaced by https://github.com/kubernetes/kubernetes/pull/95734
